### PR TITLE
cleanup task metadata

### PR DESCRIPTION
**Pull request by rishkarajgi**
## Description

Do not merge this PR until this description is replaced with relevant info.

## Related Issues

Do not merge this PR until this text is replaced with links to related issues.

## Testing

Do not merge this PR until this text is replaced with details about how these changes were tested.

## Merging

Do not merge this PR until it lists which release branches this PR should be merged / cherry-picked into.

### CODE REVIEW

File: templates/_helpers.yaml [hunk 1]
```diff
@@ -276,3 +276,89 @@ Create the name of the cleanup service account to use
...
+{{ (printf "%s/airflow.cfg" .Values.airflow.airflowHome) | quote }}
+{{- end }}
+
+
+{{ define "cleanup_task_metadata_airflow_metadata_secret" -}}
+{{ default (printf "%s-airflow-metadata" .Release.Name) .Values.airflow.data.metadataSecretName }}
+{{- end }}
+
+{{/* Standard Airflow environment variables */}}
+{{- define "cleanup_task_metadata_standard_airflow_environment" }}
```
Comment (jedcunningham) [inline templates/_helpers.yaml:291]: Why do we need a separate env?

File: templates/_helpers.yaml [hunk 2]
```diff
@@ -276,3 +276,89 @@ Create the name of the cleanup service account to use
...
 {{- end -}}
 {{- end -}}
+
+
+{{ define "cleanup_task_metadata_airflow_config_path" -}}
+{{ (printf "%s/airflow.cfg" .Values.airflow.airflowHome) | quote }}
+{{- end }}
+
+
+{{ define "cleanup_task_metadata_airflow_metadata_secret" -}}
```
Comment (jedcunningham) [inline templates/_helpers.yaml:286]: Why not use `airflow_metadata_secret`?

File: templates/_helpers.yaml [hunk 3]
```diff
@@ -276,3 +276,89 @@ Create the name of the cleanup service account to use
     {{ default "default" .Values.airflow.cleanup.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+
+{{ define "cleanup_task_metadata_airflow_config_path" -}}
```
Comment (jedcunningham) [inline templates/_helpers.yaml:281]: Why not use `airflow_config_path`?

File: values.yaml [hunk 4]
```diff
@@ -432,6 +432,14 @@ astronomer:
       repository: quay.io/astronomer/ap-certgenerator
       tag: 0.1.2
 
+cleanupTaskDataCronjob:
+  enabled: true
+  schedule: "* * * * *"
```
[the remark this answers is not in the document]
Comment (pgvishnuram) [reply values.yaml:437]: Yes - this is just for demo data purpose

File: templates/_helpers.yaml (hunk 1 above)
Comment (pgvishnuram) [reply templates/_helpers.yaml:291]: I tried to reference as it is - but external template was not accepting those env

Comment (pgvishnuram) [reply templates/_helpers.yaml]: its fixed now - removed the additional helpers added . Instead called the subchart directly

File: templates/_helpers.yaml (hunk 3 above)
Comment (pgvishnuram) [reply templates/_helpers.yaml]: its fixed now - removed the additional helpers added . Instead called the subchart directly

File: templates/_helpers.yaml (hunk 2 above)
Comment (pgvishnuram) [reply templates/_helpers.yaml]: its fixed now - removed the additional helpers added . Instead called the subchart directly

File: templates/_helpers.yaml (hunk 3 above)
Comment (pgvishnuram) [reply templates/_helpers.yaml]: ```{{ template "airflow_config_path" .Subcharts.airflow }}```  this resolved it

**Pull request closed without merging**